frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx

```diff
@@ -26,50 +26,68 @@ export const handle = {
   pageTitleI18nKey: 'communication-preference:page-title',
 } as const satisfies RouteHandleData;
 
-export async function loader({ request }: LoaderFunctionArgs) {
+export async function loader({ request, params }: LoaderFunctionArgs) {
   const { COMMUNICATION_METHOD_DIGITAL_ID } = getClientEnv();
   const preferredLanguages = await getLookupService().getAllPreferredLanguages();
   const preferredCommunicationMethods = await getLookupService().getAllPreferredCommunicationMethods();
 
-  return json({ COMMUNICATION_METHOD_DIGITAL_ID, preferredLanguages, preferredCommunicationMethods });
+  const intakeFlow = getIntakeFlow();
+  const { state } = await intakeFlow.loadState({ request, params });
+
+  return json({ digitalCommunicationMethodId: COMMUNICATION_METHOD_DIGITAL_ID, preferredLanguages, preferredCommunicationMethods, state: state.communicationPreferences });
 }
 
-export async function action({ request }: ActionFunctionArgs) {
-  const formDataSchema = z.object({
-    preferredLanguage: z.enum(['en', 'fr']),
-  });
+export async function action({ request, params }: ActionFunctionArgs) {
+  const intakeFlow = getIntakeFlow();
+  const { id } = await intakeFlow.loadState({ request, params });
+
+  const { COMMUNICATION_METHOD_DIGITAL_ID } = getClientEnv();
+
+  const emailsMatch = (val: { preferredMethod: string; email?: string; confirmEmail?: string }) => {
+    if (!val.email || !val.confirmEmail) return true;
```

Comment:
```suggestion
    if (!val.email || !val.confirmEmail) {
      // if either email is omitted, skip the matching
      // check and render 'please enter a value'
      return true;
    }
```

...only because of our slack discussion where it took me quite a while to understand this was the intent